irohad/consensus/yac/storage/yac_vote_storage.hpp

```diff
@@ -87,10 +89,21 @@ namespace iroha {
          * This parameter used on creation of proposal storage
          * @return - iter for required proposal storage
          */
-        auto findProposalStorage(const VoteMessage &msg,
-                                 PeersNumberType peers_in_round);
+        boost::optional<std::vector<YacProposalStorage>::iterator>
```

Comment:
Can't this type be generalized to support iterator to any standard collection containing `YacProposalStorage`-s, not just `vector`?

Reply:
It could be. But there is not so much reason to do it because this method is private and generalization will not lead to maintainability.